sdk/python/kfp/_client.py

```diff
@@ -1280,11 +1282,25 @@ def upload_pipeline_version(self,
 
         if pipeline_name:
             pipeline_id = self.get_pipeline_id(pipeline_name)
-
-        response = self._upload_api.upload_pipeline_version(
-            pipeline_package_path,
+        kwargs = dict(
             name=pipeline_version_name,
-            pipelineid=pipeline_id)
+            pipelineid=pipeline_id,
+        )
+
+        if description:
+            kwargs['description'] = description
+        try:
+            response = self._upload_api.upload_pipeline_version(
+                pipeline_package_path, **kwargs)
+        except kfp_server_api.exceptions.ApiTypeError as e:
+            # ToDo: Remove this once we drop support for kfp_server_api < 1.7
+            if 'description' in e.message:
+                logging.warning(
+                    'Pipeline version description is not supported in current server set up. Continuing without it.'
```

Comment:
It's easier for users to understand what they should do, if the message can be more specific. Sth along the lines of -- in current kfp-server-api pypi package.  Upgrade to x.y.z or above

Reply:
+1

Reply:
Done!